### ISSUE TITLE
test failure: restart ocid and still get pod status

### ISSUE THREAD

**Issue by mrunalp** [kubernetes-incubator/cri-o#354]
```
not ok 31 restart ocid and still get pod status
# (in test file ./pod.bats, line 283)
#   `[ "$status" -eq 0 ]' failed
# time="2017-02-03 11:06:35.968352925-08:00" level=debug msg="devicemapper: driver version is 4.35.0"
# time="2017-02-03 11:06:35.968479258-08:00" level=debug msg="devmapper: Generated prefix: container-0:42-656609"
# time="2017-02-03 11:06:35.968490880-08:00" level=debug msg="devmapper: Checking for existence of the pool container-0:42-656609-pool"
# time="2017-02-03 11:06:35.968623274-08:00" level=debug msg="devmapper: Pool doesn't exist. Creating it."
# time="2017-02-03 11:06:35.977364372-08:00" level=debug msg="devmapper: loadDeviceFilesOnStart()"
# time="2017-02-03 11:06:35.977427745-08:00" level=debug msg="devmapper: Loading data for file /tmp/tmp.oFUaoxY2SV/ocid/devicemapper/metadata/22d0c3e2f5edf0750aa44a4abc8d6f43feaa542a7972138ef4ecd5a01490701d"
# time="2017-02-03 11:06:35.977496681-08:00" level=debug msg="devmapper: Loading data for file /tmp/tmp.oFUaoxY2SV/ocid/devicemapper/metadata/8ad627f620eebe9009c1152fd7445296af724bd1b15dc24345642433e0b6fe4a"
# time="2017-02-03 11:06:35.977526605-08:00" level=debug msg="devmapper: Loading data for file /tmp/tmp.oFUaoxY2SV/ocid/devicemapper/metadata/974c87b7de086505f102e6d52614ab3fbdb3d5b1700cec49e12ef3941d6be6dc"
# time="2017-02-03 11:06:35.977551665-08:00" level=debug msg="devmapper: Loading data for file /tmp/tmp.oFUaoxY2SV/ocid/devicemapper/metadata/base"
# time="2017-02-03 11:06:35.977583817-08:00" level=debug msg="devmapper: Loading data for file /tmp/tmp.oFUaoxY2SV/ocid/devicemapper/metadata/c7e1168b3bfc3de5f05628698eb363dca16bc800966eca68d132c1d02f0bbfaa"
# time="2017-02-03 11:06:35.977625558-08:00" level=debug msg="devmapper: Loading data for file /tmp/tmp.oFUaoxY2SV/ocid/devicemapper/metadata/dd851ea16a038cc7915e36fb9c76bfceaeaed4c84bbddd1781da4a9b1ddd460e"
# time="2017-02-03 11:06:35.977666885-08:00" level=debug msg="devmapper: Skipping file /tmp/tmp.oFUaoxY2SV/ocid/devicemapper/metadata/deviceset-metadata"
# time="2017-02-03 11:06:35.977683714-08:00" level=debug msg="devmapper: Loading data for file /tmp/tmp.oFUaoxY2SV/ocid/devicemapper/metadata/e17835ba30b38902771f36bc29a0adcf8a95124d2103e08b4126e2a3d2831531"
# time="2017-02-03 11:06:35.977718319-08:00" level=debug msg="devmapper: Loading data for file /tmp/tmp.oFUaoxY2SV/ocid/devicemapper/metadata/f60f5c3dab5d615f7081916776222b417edab9649dea3f854107e477dce2f257"
# time="2017-02-03 11:06:35.977756259-08:00" level=debug msg="devmapper: Loading data for file /tmp/tmp.oFUaoxY2SV/ocid/devicemapper/metadata/f8129ccc8d535d71dd30902cfec3c9bef0181c78a4b023d57592e8540bf5fa3d"
# time="2017-02-03 11:06:35.977798775-08:00" level=debug msg="devmapper: Skipping file /tmp/tmp.oFUaoxY2SV/ocid/devicemapper/metadata/transaction-metadata"
# time="2017-02-03 11:06:35.977811862-08:00" level=debug msg="devmapper: loadDeviceFilesOnStart() END"
# time="2017-02-03 11:06:35.977820021-08:00" level=debug msg="devmapper: constructDeviceIDMap()"
# time="2017-02-03 11:06:35.977836315-08:00" level=debug msg="devmapper: Added deviceId=8 to DeviceIdMap"
# time="2017-02-03 11:06:35.977845871-08:00" level=debug msg="devmapper: Added deviceId=7 to DeviceIdMap"
# time="2017-02-03 11:06:35.977854862-08:00" level=debug msg="devmapper: Added deviceId=4 to DeviceIdMap"
# time="2017-02-03 11:06:35.977864648-08:00" level=debug msg="devmapper: Added deviceId=1 to DeviceIdMap"
# time="2017-02-03 11:06:35.977873927-08:00" level=debug msg="devmapper: Added deviceId=5 to DeviceIdMap"
# time="2017-02-03 11:06:35.977882244-08:00" level=debug msg="devmapper: Added deviceId=3 to DeviceIdMap"
# time="2017-02-03 11:06:35.977890716-08:00" level=debug msg="devmapper: Added deviceId=9 to DeviceIdMap"
# time="2017-02-03 11:06:35.977898329-08:00" level=debug msg="devmapper: Added deviceId=6 to DeviceIdMap"
# time="2017-02-03 11:06:35.977906422-08:00" level=debug msg="devmapper: Added deviceId=2 to DeviceIdMap"
# time="2017-02-03 11:06:35.977915169-08:00" level=debug msg="devmapper: constructDeviceIDMap() END"
# time="2017-02-03 11:06:35.977970450-08:00" level=warning msg="devmapper: Usage of loopback devices is strongly discouraged for production use. Please use `--storage-opt dm.thinpooldev`."
# time="2017-02-03 11:06:35.978045773-08:00" level=debug msg="devmapper: activateDeviceIfNeeded()"
# time="2017-02-03 11:06:36.003940891-08:00" level=debug msg="devmapper: UUID for device: /dev/mapper/container-0:42-656609-base is:26c1f751-72a1-43c5-9f34-57d73fdb97f8"
# time="2017-02-03 11:06:36.003966361-08:00" level=warning msg="devmapper: Base device already exists and has filesystem xfs on it. User specified filesystem  will be ignored."
# time="2017-02-03 11:06:36.003976250-08:00" level=debug msg="devmapper: deactivateDevice()"
# time="2017-02-03 11:06:36.004053391-08:00" level=debug msg="devmapper: removeDevice START(container-0:42-656609-base)"
# time="2017-02-03 11:06:36.051517000-08:00" level=debug msg="devmapper: removeDevice END(container-0:42-656609-base)"
# time="2017-02-03 11:06:36.051558363-08:00" level=debug msg="devmapper: deactivateDevice END()"
# time="2017-02-03 11:06:36.055961472-08:00" level=info msg="[graphdriver] using prior storage driver \"devicemapper\""
# ERROR: logging before flag.Parse: E0203 11:06:36.057244    2100 ocicni.go:148] error updating cni config: No networks found in /tmp/tmp.oFUaoxY2SV/cni/net.d/
# ERROR: logging before flag.Parse: W0203 11:06:36.057379    2100 ocicni.go:47] Error in finding usable CNI plugin - No networks found in /tmp/tmp.oFUaoxY2SV/cni/net.d/
# time="2017-02-03 11:06:36.060203534-08:00" level=debug msg="sandboxes: map[]"
# time="2017-02-03 11:06:36.060268860-08:00" level=debug msg="containers: &{map[] {{0 0} 0 0 0 0}}"
# time="2017-02-03 11:06:36.105811169-08:00" level=debug msg="ImageStatusRequest: image:<image:\"redis\" > "
# time="2017-02-03 11:06:36.106120646-08:00" level=debug msg="parsed reference into \"[devicemapper@/tmp/tmp.oFUaoxY2SV/ocid]docker.io/library/redis:latest\""
# time="2017-02-03 11:06:36.107256277-08:00" level=debug msg="ImageStatusResponse: image:<id:\"38da7b4ef4eae9f4fe72af4a66730fb4f7899fde13fc0a47d1ad7f4572647e8e\" repo_tags:\"docker.io/library/redis:latest\" repo_tags:\"redis\" repo_tags:\"docker://docker.io/library/redis:latest\" size:182881844 > "
# time="2017-02-03 11:06:36.148744481-08:00" level=debug msg="ImageStatusRequest: image:<image:\"redis\" > "
# time="2017-02-03 11:06:36.148881227-08:00" level=debug msg="parsed reference into \"[devicemapper@/tmp/tmp.oFUaoxY2SV/ocid]docker.io/library/redis:latest\""
# time="2017-02-03 11:06:36.149724921-08:00" level=debug msg="ImageStatusResponse: image:<id:\"38da7b4ef4eae9f4fe72af4a66730fb4f7899fde13fc0a47d1ad7f4572647e8e\" repo_tags:\"docker.io/library/redis:latest\" repo_tags:\"redis\" repo_tags:\"docker://docker.io/library/redis:latest\" size:182881844 > "
# time="2017-02-03 11:06:36.167310636-08:00" level=debug msg="RunPodSandboxRequest config:<metadata:<name:\"podsandbox1\" uid:\"redhat-test-ocid\" namespace:\"redhat.test.ocid\" attempt:1 > hostname:\"ocic_host\" log_directory:\".\" port_mappings:<protocol:UDP container_port:80 host_port:4888 host_ip:\"192.168.0.33\" > port_mappings:<protocol:2 container_port:81 host_port:4889 host_ip:\"192.168.0.33\" > labels:<key:\"group\" value:\"test\" > annotations:<key:\"owner\" value:\"hmeng\" > annotations:<key$\"security.alpha.kubernetes.io/seccomp/pod\" value:\"unconfined\" > annotations:<key:\"security.alpha.kubernetes.io/sysctls\" value:\"kernel.shm_rmid_forced=1,net.ipv4.ip_local_port_range=1024 65000\" > annotations:<key:\"security.alpha.kubernetes.io/unsafe-sysctls\" value:\"kernel.msgmax=8192\" > linux:<cgroup_parent:\"/ocid-podsandbox1\" security_context:<namespace_options:<host_network:false host_pid:false host_ipc:false > > > > "
# time="2017-02-03 11:06:36.167527321-08:00" level=debug msg="parsed reference into \"[devicemapper@/tmp/tmp.oFUaoxY2SV/ocid]docker.io/kubernetes/pause:latest\""
# time="2017-02-03 11:06:36.168126040-08:00" level=debug msg="exporting opaque data as blob \"sha256:82602d4b7498256156218400cbfdab12abb3474233c9fa0079cfe574a05afaa4\""
# time="2017-02-03 11:06:36.168558891-08:00" level=debug msg="devmapper: AddDevice(hash=6c3c284d38716523555e2e3c4d72f05a3c74854ec05c4a1bff2faeae1a935656 basehash=22d0c3e2f5edf0750aa44a4abc8d6f43feaa542a7972138ef4ecd5a01490701d)"
# time="2017-02-03 11:06:36.168644728-08:00" level=debug msg="devmapper: Failed to read /tmp/tmp.oFUaoxY2SV/ocid/devicemapper/metadata/6c3c284d38716523555e2e3c4d72f05a3c74854ec05c4a1bff2faeae1a935656 with err: open /tmp/tmp.oFUaoxY2SV/ocid/devicemapper/metadata/6c3c284d38716523555e2e3c4d72f05a3c74854ec05c4a1bff2faeae1a935656: no such file or directory"
# time="2017-02-03 11:06:36.169836633-08:00" level=debug msg="devmapper: registerDevice(10, 6c3c284d38716523555e2e3c4d72f05a3c74854ec05c4a1bff2faeae1a935656)"
# time="2017-02-03 11:06:36.170219053-08:00" level=debug msg="devmapper: AddDevice(hash=6c3c284d38716523555e2e3c4d72f05a3c74854ec05c4a1bff2faeae1a935656 basehash=22d0c3e2f5edf0750aa44a4abc8d6f43feaa542a7972138ef4ecd5a01490701d) END"
# time="2017-02-03 11:06:36.170957791-08:00" level=debug msg="created pod sandbox \"f63ec23960950f3f77cbaca44269b81659c804edc9245f3e512082a1af238034\""
# time="2017-02-03 11:06:36.171510116-08:00" level=debug msg="pod sandbox \"f63ec23960950f3f77cbaca44269b81659c804edc9245f3e512082a1af238034\" has work directory \"/tmp/tmp.oFUaoxY2SV/ocid/devicemapper-containers/f63ec23960950f3f77cbaca44269b81659c804edc9245f3e512082a1af238034/userdata\""
# time="2017-02-03 11:06:36.171656053-08:00" level=debug msg="pod sandbox \"f63ec23960950f3f77cbaca44269b81659c804edc9245f3e512082a1af238034\" has run directory \"/tmp/tmp.oFUaoxY2SV/ocid-run/devicemapper-containers/f63ec23960950f3f77cbaca44269b81659c804edc9245f3e512082a1af238034/userdata\""
# time="2017-02-03 11:06:36.187943274-08:00" level=debug msg="devmapper: activateDeviceIfNeeded(6c3c284d38716523555e2e3c4d72f05a3c74854ec05c4a1bff2faeae1a935656)"
# time="2017-02-03 11:06:36.233236553-08:00" level=debug msg="mounted container \"f63ec23960950f3f77cbaca44269b81659c804edc9245f3e512082a1af238034\" at \"/tmp/tmp.oFUaoxY2SV/ocid/devicemapper/mnt/6c3c284d38716523555e2e3c4d72f05a3c74854ec05c4a1bff2faeae1a935656/rootfs\""
# time="2017-02-03 11:06:36.329915668-08:00" level=info msg="Received container pid: 2359"
# time="2017-02-03 11:06:36.341397650-08:00" level=debug msg="RunPodSandboxResponse: pod_sandbox_id:\"f63ec23960950f3f77cbaca44269b81659c804edc9245f3e512082a1af238034\" "
# f63ec23960950f3f77cbaca44269b81659c804edc9245f3e512082a1af238034
# time="2017-02-03 11:06:36.349762341-08:00" level=debug msg="StopPodSandboxRequest pod_sandbox_id:\"f63ec23960950f3f77cbaca44269b81659c804edc9245f3e512082a1af238034\" "
# time="2017-02-03 11:06:36.349775190-08:00" level=debug msg="updating sandbox and container information"
# shutting down, got signal: Terminated
```

**Comment by mrunalp** [comment]
@nalind I am seeing this after updating to latest containers/image. Most other tests are passing for me now.
```
# time="2017-02-03 11:06:36.168558891-08:00" level=debug msg="devmapper: AddDevice(hash=6c3c284d38716523555e2e3c4d72f05a3c74854ec05c4a1bff2faeae1a935656 basehash=22d0c3e2f5edf0750aa44a4abc8d6f43feaa542a7972138ef4ecd5a01490701d)"
# time="2017-02-03 11:06:36.168644728-08:00" level=debug msg="devmapper: Failed to read /tmp/tmp.oFUaoxY2SV/ocid/devicemapper/metadata/6c3c284d38716523555e2e3c4d72f05a3c74854ec05c4a1bff2faeae1a935656 with err: open 
```

**Comment by nalind** [comment]
The "Failed to read..." message is expected when adding a device.  It's confusing, but it is expected.

**Comment by nalind** [comment]
Likely unrelated, but given that https://github.com/opencontainers/runc/blob/master/libcontainer/label/label_selinux.go#L48 checks if `con[0]` is set to `label`, should that be one of the values listed in https://github.com/opencontainers/runc/blob/master/libcontainer/label/label_selinux.go#L13?  I've gotten at least one error message about this, though it may not be the same problem that you're seeing.

**Comment by rhatdan** [comment]
@nalind label_selinux code you refereced is looking at `level` and `user`, not label?

**Comment by nalind** [comment]
@dwalsh Hmm, you're right.  I must have hallucinated that.

**Comment by nalind** [comment]
Ah, turns out I was hitting https://github.com/opencontainers/runc/pull/1312.  @mrunalp the log ends when the daemon is shut down, and there's nothing from the second copy that's started, which is where I ran into the problem in `runc`.  Was there more to the log after that?

**Comment by mrunalp** [comment]
This was fixed by #360
